spark-cassandra-connector/src/main/scala/org/apache/spark/sql/cassandra/PredicatePushDown.scala

```diff
@@ -0,0 +1,332 @@
+package org.apache.spark.sql.cassandra
+
+import com.datastax.spark.connector.NamedColumnRef
+import com.datastax.spark.connector.cql.TableDef
+import com.datastax.spark.connector.rdd.CassandraRDD
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.sources.Filter
+import org.apache.spark.sql.sources
+import org.apache.spark.sql.types.DataType
+
+class PredicatePushDown (predicates: Seq[Expression], tableDef: TableDef)
```

Comment:
This file is too big, it is named `PredicatePushDown` but this name is invalid for this file since this is not the main trait/class/object here. 
I suggest to create two files: 
1. `PushDown.scala` with `PushDown`, `PredicatePushDown` and `FilterPushDown` classes 
2. `RddSelectPdTrf.scala` with `RddSelectPdTrf`, `RddPredicateSelectPdTrf` and `RddFilterSelectPdTrf`